tcpclient/connection.go

```diff
@@ -1,14 +1,24 @@
 package tcpclient
 
-import "fmt"
+import (
+	"fmt"
+	"time"
+)
 
 type Connection struct {
-	ID     int
-	status connectionStatus
+	ID      int
+	status  connectionStatus
+	metrics connectionMetrics
```

Comment:
good idea! I like the "substruct" you are incorporating

Reply:
👍